tower-rate-limit/tests/rate_limit.rs

```diff
@@ -3,73 +3,70 @@ extern crate tower_mock;
 extern crate tower_rate_limit;
 extern crate tower_service;
 extern crate tokio_timer;
+extern crate tokio;
 
-use futures::prelude::*;
+use futures::{prelude::*, future};
 use tower_rate_limit::*;
 use tower_service::*;
 
-use std::time::Duration;
-use std::thread;
+use std::time::{Duration, Instant};
 
 #[test]
 fn reaching_capacity() {
-    let (mut service, mut handle) =
-        new_service(Rate::new(1, from_millis(100)));
+    let mut rt = tokio::runtime::Runtime::new().unwrap();
```

Comment:
For tests, I would use the current thread runtime.

You also shouldn't need `lazy` anymore with the change I suggested above.

Instead of chaining `and_then`, I would make multiple sequential calls to `block_on`.

```rust

let one = rt.block_on(...);
let two = rt.block_on(...);
```

Reply:
Ah, good point!

Reply:
@carllerche
> Instead of chaining and_then, I would make multiple sequential calls to `block_on`.
 
I'm not sure if that works, since `tower_mock::Handle::poll_request` has to be called from within a task context. 

I could block on `future::lazy(move || { handle.poll_request() })`but then `Handle` would have to be inside an `Arc<Mutex<...>>` and that doesn't seem any better...

Reply:
current_thread’s block_on shouldn’t require static. You can use poll_fn even I think.

Reply:
Ah, you're right, it that does work -- 89666da31948493d9d83b7947154d40704a78b10 should make all the changes you requested.